software/CONFIGURATION.md

```diff
@@ -22,9 +22,10 @@ default configuration:
 }
 ```
 
-- `EUROPI_MODEL` specifies the type of EuroPi module. Currently only `"europi"` is supported. Default: `"europi"`
-- `PICO_MODEL` must be one of `"pico"` or `"pico w"`. Default: `"pico"`
-- `CPU_FREQ` must be one of `250000000` or `125000000`. Default: `"250000000"`
+- `EUROPI_MODEL` specifies the type of EuroPi module. Currently only `"europi"` is supported. Default: `"europi"`.
+- `PICO_MODEL` must be one of `"pico"`, `"pico2"`, or `"pico w"`. Default: `"pico"`.
+- `CPU_FREQ` specifies whether or not the CPU should be overclocked. Must be one of `"overclocked"` or `"normal"`.
```

Comment:
[Suggestion]
Apologies if I've missed a discussion on this but is there a reason for limiting the user to only discrete values as opposed to the full range? It's a real edge case but I know some people have *under*clocked the Pico before for power consumption reasons

Reply:
I was mostly copying what was available in the original version. We probably could add a new `underclocked` option at a later date to support that. I'm not sure it's worth including in this merge request, but the framework of CPU frequencies isn't hard to expand on later if we want to add underclocking.  I have concerns about what underclocking may do to some scripts, but that's a separate question.

I'm not sure I see much benefit in allowing the full integer range of frequencies; will anyone ever realistically need to control the frequency down to the Hz? Also, specifying the actual frequency assumes the user has referred to the datasheet and knows what the supported frequencies for the processor are. The more-abstract "underclocked/normal/overclocked" option feels easier to understand to me, even though it's less informative.

Reply:
I've added `"underclocked"` as a new option, which sets the frequency to 75MHz. I haven't done a lot of testing with the lower rate, but I'd be hesitant to go much slower, since we've already been getting performance issues with some of the heavier scripts.